sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -3118,12 +3118,15 @@ def __GetBodiesFromQueryResult(result: Dict[str, Any]) -> List[Dict[str, Any]]:
 
         # check if query has prefix partition key
         isPrefixPartitionQuery = kwargs.pop("isPrefixPartitionQuery", None)
-        if isPrefixPartitionQuery:
+        if isPrefixPartitionQuery and "partitionKeyDefinition" in kwargs:
```

Comment:
why the kwargs part?

Reply:
Unlikely to be a case, but double checking helps, the code inside relies on partitionKeyDefinition being in kwargs. This only affects the sync client, since the async client handles this differently. Also stops mypy and pylint from blocking.